Add helion x pt2 test

oulgen opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #155613
* #155612
* __->__ #155513

This kinda just worked out of the box, shocking. PT2 traced into helion and emitted it as a user defined triton kernel: P1836496774

In the long run, we do not actually want this, but rather to create a helion HOP so we can do fusions etc.



cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

oulgen committed:
Add helion x pt2 test

[ghstack-poisoned]

zou3519 reviewed (approved):
I'm pretty amazed this works out of the box

drisspg commented on test/inductor/test_helion_kernels.py:
```diff
@@ -0,0 +1,49 @@
+        @helion.kernel(config=helion.Config(block_sizes=[1, 2]))
+        def add(x: torch.Tensor, y: torch.Tensor) -> torch.Tensor:
+            # match pytorch broadcasting rules
+            x, y = torch.broadcast_tensors(x, y)
```
TIL lol

oulgen committed:
None on "Add helion x pt2 test"


This kinda just worked out of the box, shocking. PT2 traced into helion and emitted it as a user defined triton kernel: P1836496774

In the long run, we do not actually want this, but rather to create a helion HOP so we can do fusions etc.



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

oulgen committed:
None on "Add helion x pt2 test"


This kinda just worked out of the box, shocking. PT2 traced into helion and emitted it as a user defined triton kernel: P1836496774

In the long run, we do not actually want this, but rather to create a helion HOP so we can do fusions etc.



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

oulgen committed:
None on "Add helion x pt2 test"


This kinda just worked out of the box, shocking. PT2 traced into helion and emitted it as a user defined triton kernel: P1836496774

In the long run, we do not actually want this, but rather to create a helion HOP so we can do fusions etc.



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

oulgen committed:
None on "Add helion x pt2 test"


This kinda just worked out of the box, shocking. PT2 traced into helion and emitted it as a user defined triton kernel: P1836496774

In the long run, we do not actually want this, but rather to create a helion HOP so we can do fusions etc.



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

oulgen commented:
@pytorchbot merge

atalman commented:
Hi @oulgen looks like this breaks triton pin update please see: https://github.com/pytorch/pytorch/pull/156186

```
6-20T21:00:08.5091389Z #55 316.4 + pip_install helion
2025-06-20T21:00:08.5091857Z #55 316.4 + as_jenkins conda run -n py_3.10 pip install --progress-bar off helion
2025-06-20T21:00:08.5093412Z #55 316.4 + sudo -E -H -u jenkins env -u SUDO_UID -u SUDO_GID -u SUDO_COMMAND -u SUDO_USER env PATH=/usr/local/nvidia/bin:/usr/local/cuda/bin:/opt/conda/envs/py_3.10/bin:/opt/conda/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin LD_LIBRARY_PATH= conda run -n py_3.10 pip install --progress-bar off helion
2025-06-20T21:00:35.3221881Z #55 343.4 ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
2025-06-20T21:00:35.3222904Z #55 343.4 timm 1.0.14 requires torchvision, which is not installed.
2025-06-20T21:00:35.3223287Z #55 343.4 
2025-06-20T21:00:35.4731146Z #55 343.4 Collecting helion
2025-06-20T21:00:35.4732552Z #55 343.4   Downloading helion-0.0.7-py3-none-any.whl.metadata (14 kB)
2025-06-20T21:00:35.4733750Z #55 343.4 Collecting filecheck (from helion)
2025-06-20T21:00:35.4734315Z #55 343.4   Downloading filecheck-1.0.2-py3-none-any.whl.metadata (5.8 kB)
2025-06-20T21:00:35.4734785Z #55 343.4 Collecting torch>=2.7.0 (from helion)
2025-06-20T21:00:35.4735296Z #55 343.4   Using cached torch-2.7.1-cp310-cp310-manylinux_2_28_x86_64.whl.metadata (29 kB)
2025-06-20T21:00:35.4736201Z #55 343.4 Requirement already satisfied: typing-extensions>=4.0.0 in /opt/conda/envs/py_3.10/lib/python3.10/site-packages (from helion) (4.14.0)
2025-06-20T21:00:35.4737344Z #55 343.4 Requirement already satisfied: filelock in /opt/conda/envs/py_3.10/lib/python3.10/site-packages (from torch>=2.7.0->helion) (3.18.0)
2025-06-20T21:00:35.4738471Z #55 343.4 Requirement already satisfied: sympy>=1.13.3 in /opt/conda/envs/py_3.10/lib/python3.10/site-packages (from torch>=2.7.0->helion) (1.13.3)
2025-06-20T21:00:35.4739641Z #55 343.4 Requirement already satisfied: networkx in /opt/conda/envs/py_3.10/lib/python3.10/site-packages (from torch>=2.7.0->helion) (2.8.8)
2025-06-20T21:00:35.4740743Z #55 343.4 Requirement already satisfied: jinja2 in /opt/conda/envs/py_3.10/lib/python3.10/site-packages (from torch>=2.7.0->helion) (3.1.6)
2025-06-20T21:00:35.4741831Z #55 343.4 Requirement already satisfied: fsspec in /opt/conda/envs/py_3.10/lib/python3.10/site-packages (from torch>=2.7.0->helion) (2025.5.1)
2025-06-20T21:00:35.4743117Z #55 343.4 Requirement already satisfied: nvidia-cuda-nvrtc-cu12==12.6.77 in /opt/conda/envs/py_3.10/lib/python3.10/site-packages (from torch>=2.7.0->helion) (12.6.77)
2025-06-20T21:00:35.4744433Z #55 343.4 Requirement already satisfied: nvidia-cuda-runtime-cu12==12.6.77 in /opt/conda/envs/py_3.10/lib/python3.10/site-packages (from torch>=2.7.0->helion) (12.6.77)
2025-06-20T21:00:35.4745689Z #55 343.4 Requirement already satisfied: nvidia-cuda-cupti-cu12==12.6.80 in /opt/conda/envs/py_3.10/lib/python3.10/site-packages (from torch>=2.7.0->helion) (12.6.80)
2025-06-20T21:00:35.4746931Z #55 343.4 Requirement already satisfied: nvidia-cudnn-cu12==9.5.1.17 in /opt/conda/envs/py_3.10/lib/python3.10/site-packages (from torch>=2.7.0->helion) (9.5.1.17)
2025-06-20T21:00:35.4748153Z #55 343.4 Requirement already satisfied: nvidia-cublas-cu12==12.6.4.1 in /opt/conda/envs/py_3.10/lib/python3.10/site-packages (from torch>=2.7.0->helion) (12.6.4.1)
2025-06-20T21:00:35.4749363Z #55 343.4 Requirement already satisfied: nvidia-cufft-cu12==11.3.0.4 in /opt/conda/envs/py_3.10/lib/python3.10/site-packages (from torch>=2.7.0->helion) (11.3.0.4)
2025-06-20T21:00:35.4750591Z #55 343.4 Requirement already satisfied: nvidia-curand-cu12==10.3.7.77 in /opt/conda/envs/py_3.10/lib/python3.10/site-packages (from torch>=2.7.0->helion) (10.3.7.77)
2025-06-20T21:00:35.4752122Z #55 343.4 Requirement already satisfied: nvidia-cusolver-cu12==11.7.1.2 in /opt/conda/envs/py_3.10/lib/python3.10/site-packages (from torch>=2.7.0->helion) (11.7.1.2)
2025-06-20T21:00:35.4753398Z #55 343.4 Requirement already satisfied: nvidia-cusparse-cu12==12.5.4.2 in /opt/conda/envs/py_3.10/lib/python3.10/site-packages (from torch>=2.7.0->helion) (12.5.4.2)
2025-06-20T21:00:35.4754675Z #55 343.4 Requirement already satisfied: nvidia-cusparselt-cu12==0.6.3 in /opt/conda/envs/py_3.10/lib/python3.10/site-packages (from torch>=2.7.0->helion) (0.6.3)
2025-06-20T21:00:35.4755920Z #55 343.4 Requirement already satisfied: nvidia-nccl-cu12==2.26.2 in /opt/conda/envs/py_3.10/lib/python3.10/site-packages (from torch>=2.7.0->helion) (2.26.2)
2025-06-20T21:00:35.4757136Z #55 343.4 Requirement already satisfied: nvidia-nvtx-cu12==12.6.77 in /opt/conda/envs/py_3.10/lib/python3.10/site-packages (from torch>=2.7.0->helion) (12.6.77)
2025-06-20T21:00:35.4758412Z #55 343.4 Requirement already satisfied: nvidia-nvjitlink-cu12==12.6.85 in /opt/conda/envs/py_3.10/lib/python3.10/site-packages (from torch>=2.7.0->helion) (12.6.85)
2025-06-20T21:00:35.4759709Z #55 343.4 Requirement already satisfied: nvidia-cufile-cu12==1.11.1.6 in /opt/conda/envs/py_3.10/lib/python3.10/site-packages (from torch>=2.7.0->helion) (1.11.1.6)
2025-06-20T21:00:35.4760566Z #55 343.4 Collecting triton==3.3.1 (from torch>=2.7.0->helion)
2025-06-20T21:00:35.4761248Z #55 343.4   Using cached triton-3.3.1-cp310-cp310-manylinux_2_27_x86_64.manylinux_2_28_x86_64.whl.metadata (1.5 kB)
2025-06-20T21:00:35.4762337Z #55 343.4 Requirement already satisfied: setuptools>=40.8.0 in /opt/conda/envs/py_3.10/lib/python3.10/site-packages (from triton==3.3.1->torch>=2.7.0->helion) (80.9.0)
2025-06-20T21:00:35.4763622Z #55 343.4 Requirement already satisfied: mpmath<1.4,>=1.1.0 in /opt/conda/envs/py_3.10/lib/python3.10/site-packages (from sympy>=1.13.3->torch>=2.7.0->helion) (1.3.0)
2025-06-20T21:00:35.4764934Z #55 343.4 Requirement already satisfied: MarkupSafe>=2.0 in /opt/conda/envs/py_3.10/lib/python3.10/site-packages (from jinja2->torch>=2.7.0->helion) (3.0.2)
2025-06-20T21:00:35.4765758Z #55 343.4 Downloading helion-0.0.7-py3-none-any.whl (149 kB)
2025-06-20T21:00:35.4766308Z #55 343.4 Using cached torch-2.7.1-cp310-cp310-manylinux_2_28_x86_64.whl (821.2 MB)
2025-06-20T21:00:35.4766984Z #55 343.4 Using cached triton-3.3.1-cp310-cp310-manylinux_2_27_x86_64.manylinux_2_28_x86_64.whl (155.6 MB)
2025-06-20T21:00:35.4767616Z #55 343.4 Downloading filecheck-1.0.2-py3-none-any.whl (23 kB)
2025-06-20T21:00:35.4768114Z #55 343.4 Installing collected packages: triton, filecheck, torch, helion
2025-06-20T21:00:35.4768566Z #55 343.4   Attempting uninstall: triton
2025-06-20T21:00:35.4768947Z #55 343.4     Found existing installation: triton 3.4.0+git76b6977d
2025-06-20T21:00:35.4769403Z #55 343.4     Uninstalling triton-3.4.0+git76b6977d:
2025-06-20T21:00:35.4769823Z #55 343.4       Successfully uninstalled triton-3.4.0+git76b6977d
2025-06-20T21:00:35.4770362Z #55 343.4 Successfully installed filecheck-1.0.2 helion-0.0.7 torch-2.7.1 triton-3.3.1
```

Please note it overwrite triton-3.4.0+git76b6977d and uses triton-3.3.1
